CRM/Event/Form/SelfSvcTransfer.php

```diff
@@ -337,7 +337,7 @@ public function postProcess() {
     }
     else {
       //cancel 'from' participant row
-      $query = "select contact_id from civicrm_email where email = '" . $params['email'] . "'";
+      $query = "select contact_id from civicrm_email left join civicrm_contact on civicrm_contact.id=civicrm_email.contact_id where email='" . $params['email'] . "' and is_deleted=0";
```

Comment:
Why left join here? Looks like the WHERE clause will ensure only civicrm_email records with a matching civicrm_contact record having is_deleted = 0 will be retrieved. And foreign key constraints should ensure you can't have a civicrm_email.contact_id that doesn't match a civicrm_contact.id . So INNER JOIN seems appropriate to me.

Also there looks to be an existing issue in the code: it's putting a parameter in a SQL query by concatenation rather than passing a parameters array to CRM_Core_DAO::executeQuery(). Would be good to fix that if altering this code, in my opinion.